File: problem_set6/readability/readability.py
from cs50 import get_string 

text = get_string("Text: ")

a = {".", "!", "?", "..."}
letter_count = 0
word_count = 1 #start from 1 to account the last word
sentence_count = 0 

for char in text:
    if char.isalpha():
        letter_count += 1
    if char == " ":
        word_count +=1
    if char in a:
        sentence_count += 1 

average_number_of_letters = (letter_count / word_count) * 100
average_number_of_words = (sentence_count / word_count) * 100

#0.0588 * L - 0.296 * S - 15.8
index = round((0.0588 * average_number_of_letters) - (0.296 * average_number_of_words) - 15.8)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else: 
    print(f"Grade {index}")

File: problem_set2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>
#include <math.h>

int main(void) {
    char *text = get_string("Text: ");
    
    int length = strlen(text);
    int letter_counter = 0;
    int sentence_counter = 0; 
    int word_counter = 0;
    double L = 0;
    double S = 0;
    double index = 0;

    for (int i = 0; i < length; i++) 
    {
        text[i] = toupper(text[i]);

        if (text[i] > 64 && text[i] < 91) {
            letter_counter++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?') {
            sentence_counter++;
        }

        if (text[i] == ' '){
            word_counter++;
        }
    }

    //ussually there is no space after last word
    word_counter++;

    // printf("letter counter %i\n", letter_counter);
    // printf("sentencecounter %i\n",sentence_counter);
    // printf("word counter %i\n", word_counter);
    
    //Coleman-Liau index
    L = (double)letter_counter / word_counter * 100; 
    S = (double)sentence_counter / word_counter * 100;
    index =  0.0588 * L - 0.296 * S - 15.8;

    printf("Grade %0.f\n", round(index));

    return 0;
}